core/src/main/scala/org/apache/spark/partial/SumEvaluator.scala

```diff
@@ -56,9 +56,12 @@ private[spark] class SumEvaluator(totalOutputs: Int, confidence: Double)
       val confFactor = {
         if (counter.count > 100) {
           new NormalDistribution().inverseCumulativeProbability(1 - (1 - confidence) / 2)
-        } else {
+        } else if (counter.count > 1) {
           val degreesOfFreedom = (counter.count - 1).toInt
           new TDistribution(degreesOfFreedom).inverseCumulativeProbability(1 - (1 - confidence) / 2)
+        } else {
+          // No way to meaningfully estimate confidence, so we signal no particular confidence interval
+          Double.PositiveInfinity
```

Comment:
Still don't think this works since you're multiplying infinity by NaN to get the intervals. It's too odd to begin with but results in NaN anyway.